scipy/interpolate/_interpolate.py

```diff
@@ -654,7 +657,8 @@ def _ensure_c_contiguous(self):
         if not self.x.flags.c_contiguous:
             self.x = self.x.copy()
         if not self.c.flags.c_contiguous:
-            self.c = self.c.copy()
+            with self._c_lock:
+                self.c = self.c.copy()
```

Comment:
Cannot comment on https://github.com/scipy/scipy/pull/20671/files#diff-54d48b9a31636ea1506371712aa1ffa21e818f270104de8068bebecb2cd3d5adL603 above: you lock on `self.c.copy()` but not on `self.c = np.moveaxis(self.c, axis+1, 0)`. Why does it not need a lock?

Reply:
The `moveaxis` call occurs in the constructor, which is guaranteed to be atomic per https://github.com/scipy/scipy/pull/20655#issuecomment-2104870366